README.md

```diff
@@ -1,32 +1,128 @@
 
-#### Package `controller`
+If you are new to Kubernetes controllers it is recommended you do not skip the any sections.
 
-The `Controller` is a simple approach that wraps the standard pattern of worker goroutines and workqueue.  The package
-depends on the `cache` and `client` removing the need for generated clientsets, informers, and listers.
+If you are lacking basic Kubernetes knowledge then it is recommended you consult other documentation on the basics first, such as the [Kubernetes Overview](https://kubernetes.io/docs/concepts/overview/).
 
-The `SharedController` type is an extension of `Controller` that adds the ability to register multiple handlers on a
-shared controller instance.
+# Kubernetes Controllers
+_Note:_ This will be a very brief crash course. We encourage you to look up other resources regarding the topic, there are many! Two recommended resources are the [Sample Controller Project](https://github.com/kubernetes/sample-controller) and [Introduction To Kubernetes Controller Frameworks](https://github.com/aiyengar2/k8s-docs/blob/main/docs/controllers/03_frameworks.md).
 
-#### Simple Example
+From [the Kubernetes Documentation](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/)
+> In Kubernetes, a controller is a control loop that watches the shared state of the cluster through the apiserver and makes changes attempting to move the current state towards the desired state.
+
+The word "controller" get used more loosely than what the above definition might have you believe. In our own words: Controllers are loop handlers that act in response to Kubernetes resource events. In Lasso, we also use the word "controller" to refer to the entire structure that is responsible for registering this loop handler and providing it the resources it needs. In short, controllers are structures that bundle everything necessary to interact with and operate on Kubernetes types.
+
+In order to achieve the above goal, a few sub-goals need to be met:
+* the controller needs to know about resource type events
+* the controller needs to be passed relevant details regarding the event to a handler
+* (optional) the controller __may__ want to interact with Kubernetes resource types
+
+There are many frameworks to help achieve the above goals. Lasso is one of them. What most of these frameworks have in common, including lasso, is that they use functionality from client-go to achieve these goals. You could achieve these goals by using just client-go. For the remainder of this section, we will fly through how a controller is put together just using client-go concepts.
+
+An informer is used to achieve the above goals needed for a Kubernetes Controller. An informer possesses an event handler and a set of indexers. The underlying structs are different but a pseudo struct is shown below to illustrate this:
+
+```
+Informer
+	EventHandler
+	Indexers
```

Comment:
I feel like the role of Indexers can get confusing without including the Store at a high level. We do not have to go into the different Store implementations but simply use Stores to help outline the relationship between Informers and Indexers. Such as, an Informer passes objects through indexers to be placed in a Store, and Indexers create indexes on those objects for efficient reads from the store.
```suggestion
Informer
	EventHandler
	Indexers
	Store
```